doc/developer/design/20231027_refresh_every_mv.md

```diff
@@ -0,0 +1,297 @@
+
+(An alternative design that would be perhaps more appealing from a conceptual point of view would be to extend refreshes infinitely both to the past and to the future from `<start_timestamp>`. However, then we couldn't really call the option `<start_timestamp>`, but would have to give it a more cryptic name, such as `<phase>`, and then we would have more explaining to do to users.)
+
+#### Refresh `ON SCHEDULE '<cron>'`
+
+To support more complex refresh schedules, we'll also support [cron schedule expressions](https://cloud.google.com/scheduler/docs/configuring/cron-job-schedules). For example, the user whose needs prompted this work would like to have daily refreshes, but not on the weekends. While this is possible to express using `REFRESH EVERY <every_interval> [STARTING AT <start_timestamp>]`, it is quite cumbersome: one would need to specify a `REFRESH EVERY '7 days'` 5 times: for each of the weekdays when a refresh should happen.
+
+The implementation for refreshes scheduled by cron expressions will be similar to `REFRESH EVERY`, except for the determination of when the next refresh should happen.
+
+#### Refresh `AT {CREATION | <timestamp>}`
+
+We perform just one refresh of the MV at the specified time. (`REFRESH AT` can be given multiple times.) Note that if the only given refresh options are all `AT` options, then there will be a last refresh, after which the MV will never be refreshed again. This will allow for certain performance optimizations in dataflows that consume the MV as input (discussed in TODO).
+
+_Before the first refresh is performed, MVs are not queryable_ (queries will block until the first refresh completes), because their since will be initially set to the time of the first refresh. The first refresh can be a significant time away from the time of creating the MV. For example, the first refresh of an MV that is updated daily could be at night, while the MV is probably created during the day. In such cases, the user would have to wait a long time to actually start using the MV. To prevent this annoying situation, users might often want to add an extra refresh at the time of creation by specifying `REFRESH AT mz_now()`. A syntactic sugar for this is `REFRESH AT CREATION`. For example,
+```SQL
+CREATE MATERIALIZED VIEW mv1 WITH (
+    REFRESH EVERY '1 day' STARTING AT '2023-11-19 03:00',
+    REFRESH AT CREATION,
+) AS SELECT ...`
+```
+
+#### Refresh `NEVER`
+
+Originally, we were planning to have a `REFRESH NEVER` option, which would perform only one refresh, at the creation time of the materialized view, and then never again. However, this is superseded by the `REFRESH AT CREATION` option. I'd say this is a better syntax than saying "never", because it's actually not "never": we do perform one refresh (at the creation time).
```

Comment:
I find this slightly confusing, for me the term "refresh" implies that the MV already has a state, and then that state gets refreshed. The `REFRESH AT CREATION` doesn't really make sense to me since there is no previous state before it is initially filled. So I vote for `REFRESH NEVER` instead. If we called it `CALCULATE AT CREATION` it would make sense on the other hand.